pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -606,6 +670,144 @@ func TestReconcile(t *testing.T) {
 		}
 	})
 
+	t.Run("Constraint with scoped enforcement actions is marked as enforced", func(t *testing.T) {
+		suffix := "ScopedMarkedEnforced"
+
+		logger.Info("Running test: Constraint is marked as enforced")
+		constraintTemplate := makeReconcileConstraintTemplate(suffix)
+		cstr := newDenyAllCstrWithScopedEA(suffix, util.AuditEnforcementPoint)
+
+		t.Cleanup(testutils.DeleteObjectAndConfirm(ctx, t, c, cstr))
+		t.Cleanup(testutils.DeleteObjectAndConfirm(ctx, t, c, expectedCRD(suffix)))
+		testutils.CreateThenCleanup(ctx, t, c, constraintTemplate)
+
+		err = retry.OnError(testutils.ConstantRetry, func(error) bool {
+			return true
+		}, func() error {
+			return c.Create(ctx, cstr)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = constraintEnforced(ctx, c, suffix)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		ns := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "testns",
+			},
+		}
+		req := admissionv1.AdmissionRequest{
+			Kind: metav1.GroupVersionKind{
+				Group:   "",
+				Version: "v1",
+				Kind:    "Namespace",
+			},
+			Operation: "Create",
+			Name:      "FooNamespace",
+			Object:    runtime.RawExtension{Object: ns},
+		}
+		resp, err := cfClient.Review(ctx, req, reviews.EnforcementPoint(util.AuditEnforcementPoint))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		gotResults := resp.Results()
+		if len(gotResults) != 1 {
+			t.Log(resp.TraceDump())
+			t.Log(cfClient.Dump(ctx))
+			t.Fatalf("want 1 result, got %v", gotResults)
+		}
+	})
+
+	t.Run("Constraint with enforcement point not supported by client is not marked as enforced", func(t *testing.T) {
+		suffix := "NotMarkedEnforced"
+
+		logger.Info("Running test: Constraint is marked as enforced")
+		constraintTemplate := makeReconcileConstraintTemplate(suffix)
+		cstr := newDenyAllCstrWithScopedEA(suffix, util.WebhookEnforcementPoint)
+
+		t.Cleanup(testutils.DeleteObjectAndConfirm(ctx, t, c, cstr))
+		t.Cleanup(testutils.DeleteObjectAndConfirm(ctx, t, c, expectedCRD(suffix)))
+		testutils.CreateThenCleanup(ctx, t, c, constraintTemplate)
+
+		err = retry.OnError(testutils.ConstantRetry, func(error) bool {
+			return true
+		}, func() error {
+			return c.Create(ctx, cstr)
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err = constraintEnforced(ctx, c, suffix)
```

Comment:
should we expect this to return err since it's not enforced?

Reply:
The name of the test was misleading, fixed the name. This is a test where `constraint` is not supposed to be part of the review because of not matching enforcement point.